PowerEditor/src/NppIO.cpp

```diff
@@ -1755,13 +1755,28 @@ bool Notepad_plus::fileRename(BufferID id)
 	if (isFileExisting)
 	{
 		CustomFileDialog fDlg(_pPublicInterface->getHSelf());
-
 		fDlg.setExtFilter(TEXT("All types"), TEXT(".*"));
+		const int langTypeIndex = setFileOpenSaveDlgFilters(fDlg, false, bufLang);
+
+		for (auto item : ScintillaEditView::langNames)
+		{
+			if (bufLang == item.LangID)
+			{
+				fDlg.setExtIndex(langTypeIndex + 1);
+				const bool defaultAllTypes = NppParameters::getInstance().getNppGUI()._setSaveDlgExtFiltToAllTypes;
+				const generic_string checkboxLabel = _nativeLangSpeaker.getLocalizedStrFromID("file-save-assign-type",
+					TEXT("&Append extension"));
+				fDlg.enableFileTypeCheckbox(checkboxLabel, !defaultAllTypes);
+				break;
+			}
+		}

```

Comment:
The loop is not necessary: `setFileOpenSaveDlgFilters` return the index of `bufLang`.